breathecode/authenticate/tests/urls/tests_github_callback.py

```diff
@@ -160,3 +160,74 @@ def test_github_callback__with_user__with_email_in_uppercase(self):
             'user_id': 1,
             'username': 'jefer94'
         }])
+
+    @mock.patch('requests.get', GithubRequestsMock.apply_get_requests_mock())
+    @mock.patch('requests.post', GithubRequestsMock.apply_post_requests_mock())
+    def test_github_callback__with_bad_user_in_querystring(self):
+        """Test /github/callback"""
+        user_kwargs = {'email': 'JDEFREITASPINTO@GMAIL.COM'}
+        role_kwargs = {'slug': 'student', 'name': 'Student'}
+        model = self.generate_models(role=True, user=True, profile_academy=True,
+            user_kwargs=user_kwargs, role_kwargs=role_kwargs)
+
+        original_url_callback = 'https://google.co.ve?user=2'
+        code = 'Konan'
+
+        url = reverse_lazy('authenticate:github_callback')
+        params = {'url': original_url_callback, 'code': code}
+        response = self.client.get(f'{url}?{urllib.parse.urlencode(params)}')
+        json = response.json()
+        expected = {
+            'detail': 'user-not-found',
+            'status_code': 404
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(self.all_user_dict(), [{
+            **self.model_to_dict(model, 'user')
+        }])
+        self.assertEqual(self.all_credentials_github_dict(), [])
+
+    @mock.patch('requests.get', GithubRequestsMock.apply_get_requests_mock())
+    @mock.patch('requests.post', GithubRequestsMock.apply_post_requests_mock())
+    def test_github_callback__with_user_in_querystring(self):
```

Comment:
We need a test in we pass through of the fake github returns a email and the user have other you should check the mock of 192 and 193, i don't know if this emails, one with lowercase and the other with uppercase match

Reply:
They match, the difference is that the one we save is Uppercase and the one sent from github is lowecarse.